src/legacy/ui/public/visualize/loader/pipeline_helpers/build_pipeline.ts

```diff
@@ -106,6 +106,25 @@ export const getSchemas = (vis: Vis, timeRange?: any): Schemas => {
       }),
     };
 
+    if (vislibCharts.includes(vis.type.name)) {
```

Comment:
can we move this into `buildVislibDimensions` function please ?

Reply:
I have to say, moving it into `buildVislibDimensions` has a drawback. This way, the `createFormat` return the wrong formatter, which later fixes by `buildVislibDimensions`. But, the later runs only for vislib visuaslizations. For example, i just found, that matric vis also use patch code for percent mode (`metric_vis_controller.js: 121`) but it's not a part of vislib, hence not passes throw buildVislibDimensions. In above approach, it's more easy to fix it just by include metric vis in the checked list.

Reply:
i am thinking about it this way:
createFormat returns a format based on the field and aggregaton. so we keep that one simple and perform only its job.

buildVisDimensions is however responsible for building dimensions for vislib specifically, and it knows that if vislib percentage mode is turned on it shouldn't use the default field formatter but always format as percentage.

if metric vis is doing the same, it should do the same, but createFormat function shouldn't just know about impementation of every visualization.

Reply:
Done